ivy/functional/frontends/tensorflow/keras/metrics.py

```diff
@@ -162,6 +162,16 @@ def kl_divergence(y_true, y_pred):
 kullback_leibler_divergence = kl_divergence
 
 
+def log_cosh(y_true, y_pred):
+    y_true = ivy.astype(y_true, y_pred.dtype, copy=False)
+    diff = y_pred - y_true
+    logval = ivy.astype(ivy.log(2.0), diff.dtype, copy=False)
+    return ivy.mean(diff + ivy.softplus(-2.0 * diff) - logval, axis=-1)
```

Comment:
Actually we have a specific function for type promotion in each frontend. So, instead of using `ivy.astype`, it would be better to import the tensorflow frontend like so: `import ivy.functional.frontends.tensorflow as tf_frontend` and then use the function `promote_types_of_tensorflow_inputs` like so: `x1, x2= tf_frontend.promote_types_of_tensorflow_inputs(x1, x2)`.

Reply:
I updated the function to use `promote_types_of_tensorflow_inputs` instead of `ivy.astype`. However, using the function with `ivy.log(2.0)` throws an assertion error in my test for dtypes float16 and float32, so I set them as unsupported. Is this okay, or should I revert it back to `ivy.astype`? Using `ivy.astype` passes the test for all dtypes.

Reply:
We should use `promote_types_of_tensorflow_inputs` to make sure the type promotion is accurate, but this seems very odd to me. I will investigate this locally and get back to you. There might be something wrong outside of your function.

Reply:
Hi @jordibardia. I see there has been a relevant fix in our main branch. Could you try and merge master in your PR?